tests/array/zipToObject.test-d.ts

```diff
@@ -0,0 +1,10 @@
+import * as _ from 'radashi'
+
+test('readonly arrays', () => {
+  const names = ['ra', 'zeus', 'loki'] as const
+  const cultures = ['egypt', 'greek', 'norse'] as const
+
+  expectTypeOf(_.zipToObject(names, cultures)).toEqualTypeOf<
+    Record<'ra' | 'zeus' | 'loki', 'egypt' | 'greek' | 'norse'>
+  >()
+})
```

Comment:
add more test cases for other types

```suggestion
import * as _ from 'radashi'
describe('zipToObject types', () => {
  test('mutable arrays', () => {
    const names = ['ra', 'zeus', 'loki']
    const cultures = ['egypt', 'greek', 'norse']
    expectTypeOf(_.zipToObject(names, cultures)).toEqualTypeOf<
      Record<string, string>
    >()
  })

  test('readonly arrays', () => {
    const names = ['ra', 'zeus', 'loki'] as const
    const cultures = ['egypt', 'greek', 'norse'] as const
    expectTypeOf(_.zipToObject(names, cultures)).toEqualTypeOf<
      Record<'ra' | 'zeus' | 'loki', 'egypt' | 'greek' | 'norse'>
    >()
  })

  test('readonly with single value', () => {
    const names = ['a', 'b'] as const
    const result = _.zipToObject(names, 1)
    expectTypeOf(result).toEqualTypeOf<
      Record<'a' | 'b', number>
    >()
  })
})
```

Reply:
Since the original error is related to types, I think it's interesting to include the other variants to ensure that everything is working.

Reply:
Thanks for this!